src/infra/accounts/inmem/src/repos/access_token_repository_inmem.rs

```diff
@@ -0,0 +1,167 @@
+                    access_token_name: access_token.token_name.clone(),
+                },
+            ));
+        }
+        if let Some(existing_token_id) = guard.token_ids_by_name.get(&access_token.token_name)
+            && guard.tokens_by_id.get(existing_token_id).is_some()
+        {
+            return Err(CreateAccessTokenError::Duplicate(
+                CreateAccessTokenErrorDuplicate {
+                    access_token_name: access_token.token_name.clone(),
+                },
+            ));
+        }
+        guard
+            .tokens_by_id
+            .insert(access_token.id, access_token.clone());
+        guard
+            .token_ids_by_name
+            .insert(access_token.token_name.clone(), access_token.id);
+        guard
+            .token_hashes_by_account_id
+            .insert(access_token.account_id.clone(), access_token.id);
+
+        Ok(())
+    }
+
+    async fn get_token_by_id(&self, token_id: &Uuid) -> Result<AccessToken, GetAccessTokenError> {
+        let guard = self.state.lock().unwrap();
+        if let Some(access_token_data) = guard.tokens_by_id.get(token_id) {
+            Ok(access_token_data.clone())
+        } else {
+            Err(GetAccessTokenError::NotFound(AccessTokenNotFoundError {
+                access_token_id: *token_id,
+            }))
+        }
+    }
+
+    async fn get_access_tokens(
+        &self,
+        pagination: &AccessTokenPaginationOpts,
+    ) -> Result<Vec<AccessToken>, GetAccessTokenError> {
+        let guard = self.state.lock().unwrap();
+
+        let access_tokens: Vec<AccessToken> = guard
+            .tokens_by_id
+            .values()
+            .skip(usize::try_from(pagination.offset).unwrap())
+            .take(usize::try_from(pagination.limit).unwrap())
+            .cloned()
+            .collect();
+
+        Ok(access_tokens)
+    }
+
+    async fn mark_revoked(
+        &self,
+        token_id: &Uuid,
+        revoked_time: DateTime<Utc>,
+    ) -> Result<(), GetAccessTokenError> {
+        let mut guard = self.state.lock().unwrap();
+        if let Some(existing_account) = guard.tokens_by_id.get(token_id) {
+            let revoked_access_token = AccessToken {
+                revoked_at: Some(revoked_time),
+                ..existing_account.clone()
+            };
+            guard.tokens_by_id.insert(*token_id, revoked_access_token);
+
+            return Ok(());
+        }
+        Err(GetAccessTokenError::NotFound(AccessTokenNotFoundError {
+            access_token_id: *token_id,
+        }))
+    }
+
+    async fn find_account_by_active_token_id(
+        &self,
+        token_id: &Uuid,
+    ) -> Result<Account, GetAccessTokenError> {
+        let access_token = self.get_token_by_id(token_id).await?;
+        if access_token.revoked_at.is_some() {
+            return Err(GetAccessTokenError::NotFound(AccessTokenNotFoundError {
+                access_token_id: *token_id,
+            }));
```

Comment:
It's not "not found", it's a different situation.

Reply:
It's a good question. I added before such an error, but after I thought that we would reveal some sensitive information by adding such a detailed error. But I agree to put it back